src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java

```diff
@@ -257,48 +292,72 @@ private void attachCaseByCcdId(
+                    ccdCaseUpdater.updateCase(
+                        exceptionRecordValidation.get(),
+                        serviceConfigItem,
+                        ignoreWarnings,
+                        idamToken,
+                        userId,
+                        targetCaseCcdRef
+                    );
+                } else {
+                    throw new CallbackException(exceptionRecordValidation.getError().mkString());
+                }
+            } else {
+                verifyExceptionRecordIsNotAttachedToCase(exceptionRecordJurisdiction, exceptionRecordId);
+
+                CaseDetails theCase = ccdApi.getCase(targetCaseCcdRef, exceptionRecordJurisdiction);
+                List<Map<String, Object>> targetCaseDocuments = getScannedDocuments(theCase);
+
+                //This is done so exception record does not change state if there is a document error
+                checkForDuplicatesOrElse(
+                    exceptionRecordDocuments,
+                    targetCaseDocuments,
+                    ids -> throwDuplicateError(targetCaseCcdRef, ids)
+                );
 
-        StartEventResponse event = ccdApi.startAttachScannedDocs(theCase, idamToken, userId);
+                List<Map<String, Object>> newCaseDocuments = attachExceptionRecordReference(
+                    exceptionRecordDocuments,
+                    exceptionRecordId
+                );
 
-        Map<String, Object> newCaseData = buildCaseData(newCaseDocuments, targetCaseDocuments);
+                StartEventResponse event = ccdApi.startAttachScannedDocs(theCase, idamToken, userId);
 
-        ccdApi.attachExceptionRecord(
-            theCase,
-            idamToken,
-            userId,
-            newCaseData,
-            createEventSummary(theCase, exceptionRecordId, newCaseDocuments),
-            event
-        );
+                Map<String, Object> newCaseData = buildCaseData(newCaseDocuments, targetCaseDocuments);
 
-        log.info(
-            "Successfully attach Exception Record with ID {} to case with id {}",
-            theCase.getId(),
-            exceptionRecord.getId()
-        );
+                ccdApi.attachExceptionRecord(
+                    theCase,
+                    idamToken,
+                    userId,
+                    newCaseData,
+                    createEventSummary(theCase, exceptionRecordId, newCaseDocuments),
+                    event
+                );
 
-        paymentsProcessor.updatePayments(exceptionRecord, theCase.getId());
+                log.info(
+                    "Successfully attach Exception Record with ID {} to case with id {}",
+                    theCase.getId(),
+                    exceptionRecordData.getId()
+                );
 
+                paymentsProcessor.updatePayments(exceptionRecordData, theCase.getId());
+            }
+        } else {
+            throw new CallbackException(classificationValidation.getError());
+        }
```

Comment:
How does the following look like for us?

```java
Classification classification = hasJourneyClassificationForAttachToCase(exceptionRecordData)
    .getOrElseThrow(() -> new CallbackException(classificationValidation.getError()));

switch (classification) {
    case EXCEPTION:
    case SUPPLEMENTARY_EVIDENCE:
        updateSupplementaryEvidence();
        break;
    case SUPPLEMENTARY_EVIDENCE_WITH_OCR:
        updateSupplementaryEvidenceWithOcr();
        break;
    case NEW_APPLICATION:
    default:
        throw new exception about invalid ocr being used here
}
```

- avoids unnecessary if/else combination together with validation tool being used here
- avoids nested if/elses
- splits functionality into methods. which should be done disregarding the suggestion

Reply:
Looks good